File: app/src/main/java/com/numero/sojodia/model/Reciprocate.java
package com.numero.sojodia.model;

import com.numero.sojodia.R;

public enum Reciprocate {
    
    GOING(
            R.string.going_to_school
    ),

    RETURN(
            R.string.coming_home
    );

    private int titleStringRes;

    Reciprocate(int titleStringRes) {
        this.titleStringRes = titleStringRes;
    }

    public int getTitleStringRes() {
        return titleStringRes;
    }

    public static Reciprocate getReciprocate(int ordinal) {
        if (Reciprocate.values().length <= ordinal) {
            return GOING;
        }
        return Reciprocate.values()[ordinal];
    }
}

File: app/src/main/kotlin/com/numero/sojodia/model/Reciprocate.kt
package com.numero.sojodia.model

import com.numero.sojodia.R

enum class Reciprocate(
        val titleStringRes: Int,
        private val shortcutValue: String
) {

    GOING(
            R.string.going_to_school,
            "going_to_school"
    ),

    RETURN(
            R.string.coming_home,
            "coming_home"
    );

    companion object {

        fun findReciprocate(position: Int): Reciprocate = if (Reciprocate.values().size <= position) {
            GOING
        } else Reciprocate.values()[position]

        fun findReciprocate(shortCut: String?): Reciprocate {
            return Reciprocate.values().find { it.shortcutValue == shortCut } ?: GOING
        }
    }
}
